### DISCUSSION TITLE
Will the Button background area get larger when hovered?

### DISCUSSION THREAD

**Discussion by Bruce9009** [AvaloniaUI/Avalonia#16178] [Q&A] [answered]
Hello,

I created a custom Button with specific `BorderBrush` and `BorderThickness` as follows. 

```
<ControlTheme x:Key="{x:Type r:Button}" TargetType="r:Button">
	<Setter Property="BorderBrush" Value="#29000000"/>
	<Setter Property="BorderThickness" Value="2"/>
	<Setter Property="Padding" Value="8"/>
	<Setter Property="CornerRadius" Value="32"/>
	<Setter Property="HorizontalAlignment" Value="Left"/>
	<Setter Property="VerticalAlignment" Value="Center"/>
	<Setter Property="HorizontalContentAlignment" Value="Center"/>
	<Setter Property="VerticalContentAlignment" Value="Center"/>
	<Setter Property="Template">
		<ControlTemplate TargetType="r:Button">
			<Border Background="{TemplateBinding Background}"
						BorderBrush="{TemplateBinding BorderBrush}"
						BorderThickness="{TemplateBinding BorderThickness}"
						CornerRadius="{TemplateBinding CornerRadius}"
						Padding="{TemplateBinding Padding}">
				<ContentPresenter Content="{TemplateBinding Content}"
								HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
							        VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}"
								Foreground="{TemplateBinding Foreground}"/>
			</Border>
		</ControlTemplate>	
	</Setter>

	<Style Selector="^:pointerover">
		<Setter Property="Background" Value="#14000000"/>
	</Style>
</ControlTheme>
```

What I would like to achieve is to just change the `Background` when hovered while keeping the `BorderBrush` and `BorderThickness`. But I found the Button background area seems to get larger when hovered, and there would be an additional border with half the original border thickness overlapping on the Button, which can be seen as follows:
![ButtonBorder](https://github.com/AvaloniaUI/Avalonia/assets/101933314/2421292c-999f-4d88-b9a1-ee6a63d64037)

I searched solutions from community and I found setting property `BackgroundSizing="InnerBorderEdge"` works. But just adding it would lead to the wrong `CornerRadius` of Button...
![ButtonBorder2](https://github.com/AvaloniaUI/Avalonia/assets/101933314/c75543b0-f189-4978-8755-8aea316d3007)

Does anyone have solution about how to resolve this question? Thanks in advance.

**Comment by timunie** [comment, marked as answer]
use a BorderBrush that is not transparent or if you are using 11.1 beta you can specify `BackgroundSizing`

**Reply by Bruce9009** [reply]
In the above example, `BorderBrush` is set as a black opacity color but it still does not work, does that mean `BorderBrush` should be neither transparent nor a color with opacity?
One more question, regarding `BackgroundSizing`, how to resolve with the wrong `CornerRadius` when adding `BackgroundSizing`?

**Reply by timunie** [reply]
Yes, it means no opacity in color of BorderBrush. Instead measure the gray tone and set it explicitly
